src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -586,6 +600,170 @@ def image_mask_iterator():
+
+    # Filter empty data
+    traces_to_add = {trace_name: trace for trace_name, trace in traces_to_add.items() if trace is not None}
+    # Filter all zero data
+    traces_to_add = {
+        trace_name: trace for trace_name, trace in traces_to_add.items() if any(x != 0 for x in np.ravel(trace))
+    }
+
+    # Early return if there is nothing to add
+    if not traces_to_add:
+        return nwbfile
+
+    # Create a reference for ROIs from the plane segmentation
+    roi_table_region = _create_roi_table_region(
+        segmentation_extractor=segmentation_extractor,
+        nwbfile=nwbfile,
+        metadata=metadata_copy,
+        plane_index=plane_index,
+    )
+
+    rate = (
+        np.nan
+        if segmentation_extractor.get_sampling_frequency() is None
+        else float(segmentation_extractor.get_sampling_frequency())
+    )
+
+    roi_response_series_kwargs = dict(
+        rois=roi_table_region,
+        # TODO: timestamps or rate (for timestamps we need frame_to_time method)
+        # TODO: check for regularity, only use timestamps if irregular timing
+        starting_time=0.0,
+        rate=rate,
+        unit="n.a.",
+    )
+
+    container = defaultdict(
+        lambda: _get_segmentation_data_interface(
+            nwbfile=nwbfile,
+            data_interface_name=fluorescence_name,
+        ),
+        # we expect that the df/F trace name is "Dff"
+        Dff=_get_segmentation_data_interface(
+            nwbfile=nwbfile,
+            data_interface_name=dff_name,
+        ),
+    )
+
+    for trace_name, trace in traces_to_add.items():
+        # Extract the response series metadata
+        trace_name = "RoiResponseSeries" if trace_name == "raw" else trace_name.capitalize()
+        trace_name = trace_name if plane_index == 0 else trace_name + f"_Plane{plane_index}"
+
+        if trace_name in container[trace_name].roi_response_series:
+            continue
+
+        metadata = dff_metadata if trace_name == "Dff" else fluorescence_metadata
+        response_series_metadata = metadata["roi_response_series"]
+        trace_metadata = next(
+            trace_metadata for trace_metadata in response_series_metadata if trace_name == trace_metadata["name"]
+        )
+
+        # Build the roi response series
+        roi_response_series_kwargs.update(
+            data=np.array(trace).T,
+            rois=roi_table_region,
+            **trace_metadata,
+        )
+        roi_response_series = RoiResponseSeries(**roi_response_series_kwargs)
+
+        # Add it to the container
+        container[trace_name].add_roi_response_series(roi_response_series)
+
+    return nwbfile
+
+
+def _create_roi_table_region(
+    segmentation_extractor: SegmentationExtractor,
+    nwbfile: NWBFile,
+    metadata: dict,
+    plane_index: int,
+):
+    """Private method to create ROI table region."""
+    add_plane_segmentation(segmentation_extractor=segmentation_extractor, nwbfile=nwbfile, metadata=metadata)
```

Comment:
This is only making sure that there exists a plane segmentation (if not create one with that call) where we can reference the ROIs. We are not passing the compression/iterator options here though.